core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -161,6 +161,31 @@ convert_errno(int error)
     return errors[error];
 }
 
+static bool
+ns_lookup_list_search(char **list, const char *host)
+{
+    size_t host_len = strlen(host), suffix_len;
+
+    while (*list) {
```

Comment:
There might be an issue, like "access out of boundary". 
`list++` might access some context behind `list` and not NULL (pass the check `*list`).

Reply:
The list is guaranteed to end with `NULL`, see [this code](https://github.com/bytecodealliance/wasm-micro-runtime/commit/0f7dc0a38650da77c1ad333f685274894b2edc9f#diff-bfd0d83ff0f2d993326c15717688c3abb59d6f2384044769168b4316e0b6ea0bR2430)